File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy=prices[0],profit=0;
        for(int i=1;i<prices.size();i++)
        {
            if(buy>prices[i])
                buy=prices[i];
            int pist=prices[i]-buy;
            if(pist>profit)
                profit=pist;
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l =0, r=0;
        int maxProfit = 0;
        while (r<prices.length) {
            if (prices[l] < prices[r]) {
                maxProfit = Math.max(maxProfit, (prices[r]-prices[l]));
            } else {
                l = r;
            }
            r++;
        }
        return maxProfit;
    }
}